### ISSUE TITLE
CLI app to run line/grid intersection

### ISSUE THREAD

**Issue by tomalrussell** [nismod/snail#8]
Read a raster file and a vector file with LineString geometries, write the splits to a new vector file.

Suggested interface:
```
./snail_split --raster flood.tif --vector roads.gpkg --output road_splits.gpkg
```

Maybe need to specify `--output-format` as well as filename to set up writer.

Only need metadata from raster, really. Could have an option to specify the raster parameters directly (transform, grid size).

Implementation options to consider:
- all-cpp: link against [GDAL](https://gdal.org/api/index.html#id3) to read and write common formats
- python wrapper: [pybind11](https://pybind11.readthedocs.io/en/stable/) interface to C++ library then handle interface and CLI through python, has the bonus of giving a python interface too.

EDIT: Pybind11 route, see subsissue #14

**Comment by tomalrussell** [comment]
- [geopandas](https://geopandas.org/) Python, [PyGEOS](https://github.com/pygeos/pygeos) wraps geometry column (to check implementation/integration details.

**Comment by tomalrussell** [comment]
Closed by #17
